inc/ipaddress.class.php

```diff
@@ -1047,112 +1049,166 @@ static function getHTMLTableCellsForItem(HTMLTableRow $row = null, CommonDBTM $i
-                        FROM `glpi_ipaddresses_ipnetworks` AS LINK
-                        JOIN `glpi_ipaddresses` AS ADDR ON (ADDR.`id` = LINK.`ipaddresses_id`
-                                                            AND ADDR.`itemtype` != 'NetworkName'
-                                                            AND ADDR.`is_deleted` = 0)
-                        LEFT JOIN `glpi_entities` ON (ADDR.`entities_id` = `glpi_entities`.`id`)
-                        WHERE LINK.`ipnetworks_id` = '".$item->getID()."')";
-
-         $query = implode('UNION ', $queries);
+         $criteria = $main_criteria;
+         $criteria['SELECT'] = array_merge($criteria['SELECT'], [
+            'NAME.id AS name_id',
+            'PORT.id AS port_id',
+            new \QueryExpression('NULL AS ' . $DB->quoteName('item_id')),
+            new \QueryExpression("NULL AS " . $DB->quoteName('item_type')),
+         ]);
+         $criteria['INNER JOIN'] = $criteria['INNER JOIN'] + [
+            'glpi_networknames AS NAME'   => [
+               'ON' => [
+                  'NAME'   => 'id',
+                  'ADDR'   => 'items_id', [
+                     'AND' => [
+                        'NAME.itemtype' => 'NetworkPort'
+                     ]
+                  ]
+               ]
+            ],
+            'glpi_networkports AS PORT'   => [
+               'ON' => [
+                  'NAME'   => 'items_id',
+                  'PORT'   => 'id', [
+                     'NOT' => [
+                        'PORT.itemtype' => $CFG_GLPI['networkport_types']
+                     ]
+                  ]
+               ]
+            ]
+         ];
+         $queries[] = $criteria;
+
+         $criteria = $main_criteria;
+         $criteria['SELECT'] = array_merge($criteria['SELECT'], [
+            'NAME.id AS name_id',
+            new \QueryExpression("NULL AS " . $DB->quoteName('port_id')),
+            new \QueryExpression('NULL AS ' . $DB->quoteName('item_id')),
+            new \QueryExpression("NULL AS " . $DB->quoteName('item_type'))
+         ]);
+         $criteria['INNER JOIN'] = $criteria['INNER JOIN'] + [
+            'glpi_networknames AS NAME'   => [
+               'ON' => [
+                  'NAME'   => 'id',
+                  'ADDR'   => 'items_id', [
+                     'AND' => [
+                        'NAME.itemtype' => ['!=', 'NetworkPort']
+                     ]
+                  ]
+               ]
+            ]
+         ];
+         $queries[] = $criteria;
+
+         $criteria = $main_criteria;
+         $criteria['SELECT'] = $criteria['SELECT'] + [
+            new \QueryExpression("NULL AS name_id"),
+            new \QueryExpression("NULL AS port_id"),
+            new \QueryExpression('NULL AS item_id'),
+            new \QueryExpression("NULL AS item_type")
+         ];
+         $criteria['INNER JOIN']['glpi_ipaddresses AS ADDR']['ON'][0]['AND']['ADDR.itemtype'] = ['!', 'NetworkName'];
```

Comment:
Maybe redfining the whole `$criteria['INNER JOIN']['glpi_ipaddresses AS ADDR']['ON']` whould be more maintainable/readable.

Reply:
Goal was to change very few things in the array... But I do not like none of those solutions :D